File: macosApp/macosApp/Log.swift
import Foundation
import ComposeAppMac

/// Single point of entry for diagnostic logging from Swift. Forwards to the
/// shared Kotlin formatter (`LogBridgeKt.macosLog`) so timestamps and the
/// stdout/stderr redirect to `~/Library/Logs/KAltSwitch.log` are configured
/// in one place. Swift call sites stay unchanged: `log("[swift] ...")`.
func log(_ message: String) {
    LogBridgeKt.macosLog(message: message)
}

File: sharedUI/src/commonMain/kotlin/com/shish/kaltswitch/log/Log.kt
package com.shish.kaltswitch.log

/**
 * Single point of entry for diagnostic event logging from common code.
 *
 * The implementation is platform-specific: on macOS it prepends a wall-clock
 * timestamp at the moment the call is made and writes to stdout (which the
 * Swift host redirects to ~/Library/Logs/KAltSwitch.log); other platforms
 * (notably commonTest on macosArm64) get the same behaviour. Capturing
 * time *here* — not at the event source — keeps callers free of clock
 * plumbing.
 *
 * Format expected:
 *   `2026-04-29 17:52:03.991 [ctl] message`
 */
expect fun log(message: String)
